modules/related-posts/jetpack-related-posts.php

```diff
@@ -246,6 +251,118 @@ public function get_target_html_unsupported() {
+			'show_context' => isset( $attributes['displayContext'] ) ? (bool) $attributes['displayContext'] : true,
+			'layout' => isset( $attributes['postLayout'] ) && $attributes['postLayout'] === 'list' ? $attributes['postLayout'] : 'grid',
+			'size' => ! empty( $attributes['postsToShow'] ) ? absint( $attributes['postsToShow'] ) : 3,
+		);
+
+		$related_posts = $this->get_for_post_id( get_the_ID(), array(
+			'size' => $block_attributes['size'],
+		) );
+
+		if ( ! $related_posts ) {
+			return '';
+		}
+
+		ob_start();
+		?>
+		<div id="jp-relatedposts" class="jp-relatedposts" style="display: block;">
+			<div class="jp-relatedposts-items jp-relatedposts-items-visual jp-relatedposts-<?php echo esc_attr( $block_attributes['layout'] ); ?>">
+				<?php foreach ( $related_posts as $index => $related_post ): 
+					$classes = array_filter( array(
+						'jp-relatedposts-post',
+						'jp-relatedposts-post' . $index,
+						! empty( $block_attributes['show_thumbnails'] ) ? 'jp-relatedposts-post-thumbs' : '',
+					) );
+					?>
+					<div
+						class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>"
+						data-post-id="<?php echo esc_attr( $related_post['id'] ); ?>"
+						data-post-format="<?php echo esc_attr( ! empty( $related_post['format'] ) ? $related_post['format'] : 'false' ); ?>"
+					>
+						<?php if ( ! empty( $block_attributes['show_thumbnails'] ) && ! empty( $related_post['img']['src'] ) ): ?>
+							<a class="jp-relatedposts-post-a"
+								href="<?php echo esc_url( $related_post['url'] ); ?>"
+								title="<?php echo esc_attr( $related_post['title'] ); ?>"
+								rel="<?php echo esc_attr( $related_post['rel'] ); ?>"
+								data-origin="<?php echo esc_attr( $related_post['url_meta']['origin'] ); ?>"
+								data-position="<?php echo esc_attr( $related_post['url_meta']['position'] ); ?>"
+							>
+								<img class="jp-relatedposts-post-img"
+									src="<?php echo esc_url( $related_post['img']['src'] ); ?>"
+									width="<?php echo esc_attr( $related_post['img']['width'] ); ?>"
+									alt="<?php echo esc_attr( $related_post['title'] ); ?>"
+								/>
+							</a>
+						<?php endif; ?>
+
+						<h4 class="jp-relatedposts-post-title">
+							<a
+								class="jp-relatedposts-post-a"
+								href="<?php echo esc_url( $related_post['url'] ); ?>"
+								title="<?php echo esc_attr( $related_post['title'] ); ?>"
+								rel="<?php echo esc_attr( $related_post['rel'] ); ?>"
+								data-origin="<?php echo esc_attr( $related_post['url_meta']['origin'] ); ?>"
+								data-position="<?php echo esc_attr( $related_post['url_meta']['position'] ); ?>"
+							>
+								<?php echo esc_html( $related_post['title'] ); ?>
+							</a>
+						</h4>
+
+						<p class="jp-relatedposts-post-excerpt"><?php echo esc_html( $related_post['excerpt'] ); ?></p>
+
+						<?php if ( $block_attributes['show_date'] ): ?>
+							<p class="jp-relatedposts-post-date" style="display: block;">
```

Comment:
There are so many missed opportunities for structured data here.

Just flagging this because I know it's something @davidakennedy was thinking about.

pafL3P-2b-p2

Reply:
Good idea! Let's consider this, but in a different, separate effort which would target both the block and the legacy Related Posts markup.